docs/writing-tests/reftest-tutorial.md

```diff
@@ -0,0 +1,262 @@
+summary:
+
+```diff
+ <!DOCTYPE html>
+ <meta charset="utf-8">
+ <title>BDO element dir=rtl</title>
+ <link rel="author" title="Sam Smith" href="mailto:sam@example.com">
+ <link rel="help" href="https://html.spec.whatwg.org/#the-bdo-element">
++<link rel="match" href="rtl-ref.html">
+ <meta name="assert" content="BDO element's DIR content attribute renders corrently given value of 'rtl'.">
+
+ <p>Test passes if WAS is displayed below.</p>
+ <bdo dir="rtl">SAW</bdo>
+```
+
+Now, anyone (human or computer) reviewing the test file will know where to find
+the associated reference file.
+
+## Verifying our work
+
+We're done writing the test, but we should make sure it fits in with the rest
+of WPT before we submit it. This involves using some of the project's tools, so
+this is the point you'll need to [configure your system to run
+WPT](../running-tests/from-local-system).
+
+[The lint tool](lint-tool) can detect some of the common mistakes people make
+when contributing to WPT. To run it, open a command-line terminal, navigate to
+the root of the WPT repository, and enter the following command:
+
+    python ./wpt lint html/semantics/text-level-semantics/the-bdo-element
+
+If this recognizes any of those common mistakes in the new files, it will tell
+you where they are and how to fix them. If you do have changes to make, you can
+run the command again to make sure you got them right.
+
+Now, we'll run the test using the automated pixel-by-pixel comparison approach
+mentioned earlier. This is important for reftests because the test and the
+reference may differ in very subtle ways that are hard to catch with the naked
+eye. That's not to say your test has to pass in all browsers (or even in *any*
+browser). But if we expect the test to pass, then running it this way will help
+us catch other kinds of mistakes.
+
+The tools support running the tests in many different browsers. We'll use
+Firefox this time:
+
+    python ./wpt run firefox html/semantics/text-level-semantics/the-bdo-element/rtl.html
+
+We expect this test to pass, so if it does, we're ready to submit it. If we
+were testing a web platform feature that Firefox didn't support, we would
+expect the test to fail instead.
+
+There are a few problems to look out for in addition to passing/failing status.
+The report will describe fewer tests than we expect if the isn't run at all.
+That's usually a sign of a formatting mistake, so you'll want to make sure
+you've used the right file names and metadata. Separately, the web browser
+might crash. That's often a sign of a browser bug, so you should consider
+reporting it to the browser's maintainers!
+
+## Submitting the test
+
+First, let's stage the new files for committing:
+
+    $ git add html/semantics/text-level-semantics/the-bdo-element/rtl.html
+    $ git add html/semantics/text-level-semantics/the-bdo-element/rtl-ref.html
+
+We can make sure the commit has everything we want to submit (and nothing we
+don't) using `git diff`:
```

Comment:
```suggestion
don't), using `git diff`:
```

Reply:
This matches the way I would enunciate the sentence, but I don't see how it's grammatically necessary. If the parenthetical weren't there, then the comma would feel more out of place.

Though scrutinizing the sentence along those lines makes me think that the word "by" is necessary here.

Reply:
Actually to me, if it had *“We can make sure the commit has everything we want to submit using `git diff`”*, it’s even more ambiguous: Does it mean there is some set of things you want to use `git diff` to submit? Or does it instead mean, *Using `git diff`, we can make sure the commit has everything we want to submit.*? Of course it means the latter, but without the comma — or without moving the *Using `git diff`* to the beginning of the sentence instead of the end — it’s ambiguous. But I don’t feel strongly about it needing to be changed. I defer to your judgment on  whether you think it’s fine as-is. Like a lot of copy edits, it’s just a judgment call.